### PR TITLE
gogctuner: fix unstable test in the TestTuner

### LINKED ISSUE #38467: unstable test: TestTuner
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```shell
=== RUN   TestTuner
    tuner_test.go:56: 
        	Error Trace:	/home/jenkins/.tidb/tmp/5b8829669b2a941b840f9eb369e2b54f/sandbox/processwrapper-sandbox/4060/execroot/__main__/bazel-out/k8-fastbuild/bin/util/gctuner/gctuner_test_/gctuner_test.runfiles/__main__/util/gctuner/tuner_test.go:56
        	Error:      	"20" is not greater than or equal to "50"
        	Test:       	TestTuner
--- FAIL: TestTuner (0.50s)
```
link  https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-hourly/detail/tidb-unit-test-hourly/9340/tests

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #38467

Problem Summary:

### What is changed and how it works?

Now the default min value of GOGC is 100 for gogc tuner.  so This range in test is too small to run.

the origin threshold is so small for the test case to be disturbed by the memory of the test framework.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
    Run the test 100 times
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: util/gctuner/tuner_test.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ func TestTuner(t *testing.T) {
 	runtime.GC()
 	for i := 0; i < 100; i++ {
 		runtime.GC()
-		require.GreaterOrEqual(t, tn.getGCPercent(), uint32(50))
-		require.LessOrEqual(t, tn.getGCPercent(), uint32(100))
+		require.GreaterOrEqual(t, tn.getGCPercent(), MinGCPercent)
+		require.LessOrEqual(t, tn.getGCPercent(), uint32(200))
```
Comment (lance6716) [inline util/gctuner/tuner_test.go:57]: This should not change? the active heap can be equal or larger than threshold/2, so we should expect the percentage equal or less than 100%.

Also, this test have repeated many hardcode values.

**Review by lance6716 — COMMENTED** [review]
> Now the default min value of GOGC is 100 for gogc tuner

Why we do this? I'm afraid it will increase the possibility of OOM.

**Comment by hawkingrei** [comment]
> > Now the default min value of GOGC is 100 for gogc tuner
> 
> Why we do this? I'm afraid it will increase the possibility of OOM.

At the earliest, the default value of gogctuner is less than 100. the main reason is the global memory limiter needs to use GOGC to judge the time of forcely garbage collection. so we need to change our strategy. GOGC tuner now has a max memory threshold. if memory is over it, gogc tuner will stop working. this threshold is always less than the threshold of global memory limiter.

File: util/gctuner/tuner_test.go (hunk 1 above)
Comment (hawkingrei) [reply util/gctuner/tuner_test.go]: updated

Comment (hawkingrei) [reply util/gctuner/tuner_test.go]: the origin threshold is so small for the test case to be disturbed by the memory of the test framework.

**Review by Yisaer — APPROVED** [review]

**Pull request merged**